lib/irb/color.rb

```diff
@@ -0,0 +1,96 @@
+# frozen_string_literal: true
+require 'ripper'
+
+module IRB # :nodoc:
+  module Color
+    CLEAR     = 0
+    BOLD      = 1
+    UNDERLINE = 4
+    RED       = 31
+    GREEN     = 32
+    BLUE      = 34
+    MAGENTA   = 35
+    CYAN      = 36
+
+    TOKEN_KEYWORDS = {
+      on_kw: ['nil', 'self', 'true', 'false'],
+      on_const: ['ENV'],
+    }
+
+    TOKEN_SEQ_EXPRS = {
+      on_CHAR:            [[BLUE, BOLD],            [Ripper::EXPR_END]],
+      on_const:           [[BLUE, BOLD, UNDERLINE], [Ripper::EXPR_ARG, Ripper::EXPR_CMDARG]],
+      on_embexpr_beg:     [[RED],                   [Ripper::EXPR_BEG, Ripper::EXPR_END]],
+      on_embexpr_end:     [[RED],                   [Ripper::EXPR_END, Ripper::EXPR_ENDFN]],
+      on_ident:           [[BLUE, BOLD],            [Ripper::EXPR_ENDFN]],
+      on_int:             [[BLUE, BOLD],            [Ripper::EXPR_END]],
+      on_float:           [[MAGENTA, BOLD],         [Ripper::EXPR_END]],
+      on_kw:              [[GREEN],                 [Ripper::EXPR_CLASS, Ripper::EXPR_BEG, Ripper::EXPR_END, Ripper::EXPR_FNAME]],
+      on_label:           [[MAGENTA],               [Ripper::EXPR_LABELED]],
+      on_qwords_beg:      [[RED],                   [Ripper::EXPR_BEG]],
+      on_regexp_beg:      [[RED],                   [Ripper::EXPR_BEG]],
+      on_regexp_end:      [[RED],                   [Ripper::EXPR_BEG]],
+      on_symbeg:          [[BLUE, BOLD],            [Ripper::EXPR_FNAME]],
+      on_tstring_beg:     [[RED],                   [Ripper::EXPR_BEG, Ripper::EXPR_END, Ripper::EXPR_ARG, Ripper::EXPR_CMDARG]],
+      on_tstring_content: [[RED],                   [Ripper::EXPR_BEG, Ripper::EXPR_ARG, Ripper::EXPR_CMDARG]],
+      on_tstring_end:     [[RED],                   [Ripper::EXPR_END]],
+    }
+
+    class << self
+      def colorable?
+        $stdout.tty? && ENV.key?('TERM')
```

Comment:
```suggestion
        $stdout.tty? && ENV.key?('TERM') && ENV['TERM'] != 'dumb'
```
Should we check TERM value is not `dumb`?

Reply:
good catch. Thanks!